src/main/java/task01/Main.java

```diff
@@ -0,0 +1,22 @@
+package task01;
+
+import java.time.LocalDate;
+import java.util.Objects;
+import java.util.Optional;
+
+public class Main {
+
+    public static void main(String[] args) {
+
+        // work with records
+//        var person = new task01.Person("Антон", "Эста", Optional.of("Викторович"), LocalDate.of(1985, 5, 10), "Красноярск-45");
+        var person = new Person("Антон", "Эста", Optional.empty(), LocalDate.of(1985, 5, 10), "Красноярск-45");
+        var insurance = new Insurance(12687894523L, person);
+        insurance.printMandatory();
+
+        // work with NPE
+        Objects.requireNonNull(null);
```

Comment:
Ну так, наверное, действительно ничего не увидим.
Думаю, имелось в виду что-то вроде
```
public class M1 {
    static class Car {
        private Engine engine;

        public Car(Engine engine) {
            this.engine = engine;
        }

        public Engine getEngine() {
            return engine;
        }
    }
    static class Engine {
        private Double powerful;

        public Engine(Double powerful) {
            this.powerful = powerful;
        }

        public Double getPowerful() {
            return powerful;
        }
    }
    public static void main(String[] args) {
        final Car car = new Car(new Engine(null));
        System.out.println("isRocketEngine = " + (car.getEngine().getPowerful() > 1000));
    }
}
```
До 11 джавы информации будет негусто
```
Exception in thread "main" java.lang.NullPointerException
	at ru.geekbrains.june.chat.client.M1.main(M1.java:30)
```

Начиная с 11 побольше
```
Exception in thread "main" java.lang.NullPointerException: Cannot invoke "java.lang.Double.doubleValue()" because the return value of "ru.geekbrains.june.chat.client.M1$Engine.getPowerful()" is null
	at ru.geekbrains.june.chat.client.M1.main(M1.java:30)
```